### PR TITLE
Try to simplify server method definition

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Also trying to get my feet wet with this change, nothing too fancy.

### CODE REVIEW

File: core/src/main/java/io/grpc/ServerServiceDefinition.java [hunk 1]
```diff
@@ -60,21 +60,19 @@ public String getName() {
     return name;
   }
 
-  public ImmutableList<ServerMethodDefinition<?, ?>> getMethods() {
-    return methods;
+  public ImmutableCollection<ServerMethodDefinition<?, ?>> getMethods() {
```
Comment (nmittler) [inline core/src/main/java/io/grpc/ServerServiceDefinition.java]: Maybe just return `Collection`?

**Comment by nmittler** [comment]
LGTM ... @ejona86 do you want to take a look?

File: core/src/main/java/io/grpc/ServerServiceDefinition.java (hunk 1 above)
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/ServerServiceDefinition.java]: Done.

**Comment by zhangkun83** [comment]
This collides with the first part of #612 , while I am doing less than this PR. I will let you reviewers decide which version to take. I can easily drop my first commit if we go with this PR.

File: core/src/test/java/io/grpc/ServerInterceptorsTest.java [hunk 2]
```diff
@@ -51,6 +50,8 @@
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
 
+import io.grpc.Metadata.Headers;
```
Comment (ejona86) [inline core/src/test/java/io/grpc/ServerInterceptorsTest.java]: io is sorted above org; the previous location was correct.

**Comment by ejona86** [comment]
@zhangkun83, the two changes are mostly compatible. It looks like just the tests would cause trouble. These changes are beneficial and a sort of additional cleanup to the ones you made. Could you and @carl-mastrangelo discuss how you want to deal with the conflicts?

File: core/src/test/java/io/grpc/MutableHandlerRegistryImplTest.java [hunk 3]
```diff
@@ -38,14 +39,14 @@
...
 
-import io.grpc.HandlerRegistry.Method;
-
 import org.junit.After;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 import org.mockito.Mockito;
 
+import io.grpc.HandlerRegistry.Method;
```
Comment (ejona86) [inline core/src/test/java/io/grpc/MutableHandlerRegistryImplTest.java]: This also in the correct spot before.

**Comment by ejona86** [comment]
@carl-mastrangelo LGTM

**Comment by carl-mastrangelo** [comment]
Fixed the imports (Eclipse sorts funny sometimes).  Kun said my commit was doing more so I should submit.  @ejona86 PTAL.

**Comment by ejona86** [comment]
@carl-mastrangelo, still LGTM. We tend to rebase on top of master and squash any useless commits before pushing to master. I can help you out with that process.

**Comment by ejona86** [comment]
Also, note that Travis is failing for your change.

**Pull request merged**